rasterio/_io.pyx

```diff
@@ -2152,9 +2170,17 @@ def writer(path, mode, **kwargs):
             "VFS '{0}' datasets can not be created or updated.".format(
                 scheme))
 
+    # These drivers are known to produce invalid results with
+    # IndirectRasterUpdater. Save users the trouble and fail fast.
+    bad_write_drivers = ("netCDF", )
```

Comment:
@perrygeo this should probably be a global variable, in ALL_CAPS toward the top of this file, to make it really obvious (so that we can add other drivers to it later).